x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -743,6 +814,46 @@ export const getGroupedESQuery = (
   }
 };
 
+const getContextAggregation = (params: any) => {
+  const validPrefixForContext = ['host', 'cloud', 'orchestrator', 'container', 'labels', 'tags'];
+  const positiveComparators = getPositiveComparators();
+  const positiveCriteria = params.criteria.filter((criterion: any) =>
+    positiveComparators.includes(criterion.comparator)
+  );
+
+  const includesFromGroupBy = params.groupBy
+    ? getIncludeSet(params.groupBy, validPrefixForContext)
+    : new Set<string>();
+  const includesFromCriteria = getIncludeSet(
+    positiveCriteria.map((criterion: any) => criterion.field),
+    validPrefixForContext
+  );
+  const includesList = Array.from(
+    new Set<string>([...includesFromGroupBy, ...includesFromCriteria])
+  );
+
+  const excludesList = includesList.includes('host')
+    ? ['host.cpu.*', 'host.disk.*', 'host.network.*']
+    : [];
+
+  const additionalContextAgg =
+    includesList.length > 0
+      ? {
+          additionalContext: {
+            top_hits: {
+              size: 1,
+              _source: {
```

Comment:
So far the Logs UI code aims to avoid `_source` in favor of `fields`. That way it fully supports runtime fields and synthetic `_source` when accessing log data. (There are exceptions for internal ML indices, for which the user can't define runtime fields anyway.)

 Do you see any way of preserving these? Is `fields` inside of `top_hits` or `top_metrics` supported maybe?

Reply:
I have replaced `_source` with `fields` in `top_hits`.